File: Flutter frontend/practise2/design/lib/Service.dart
import 'Professional.dart';

class Service {
  final int id;
  final String nom;
  final String? description;
  final String? categorie;
  final List<Professional> professionnels;
  final String service_photo;

  Service({
    required this.id,
    required this.nom,
    this.description,
    this.categorie,
    required this.professionnels,
    required this.service_photo,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      id: json['id'] as int,
      nom: json['nom'] as String,
      description:
          json['description'] != null ? json['description'] as String : '',
      categorie: json['categorie'] != null ? json['categorie'] as String : '',
      professionnels: (json['professionnels'] as List?)
              ?.map((proJson) => Professional.fromJson(proJson))
              .toList() ??
          [],
      service_photo: json['service_photo'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nom': nom,
      'description': description,
      'categorie': categorie,
      'professionnels': professionnels.map((pro) => pro.toJson()).toList(),
      'service_photo': service_photo,
    };
  }
}

File: spring backend/sosMaison/sosMaison/src/main/java/com/sosMaison/sosMaison/Services/Service.java
package com.sosMaison.sosMaison.Services;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.sosMaison.sosMaison.Localisation.Localisation;
import com.sosMaison.sosMaison.Professional.Professional;
import jakarta.persistence.*;
import lombok.*;

import java.util.*;
@Setter
@Getter
@Entity
@Data



public class Service {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nom;
    private String description;
    private String categorie;

    @OneToMany(mappedBy = "service")
    @JsonManagedReference
    private List<Professional> professionnels;

   private String service_photo;

    public Service(Long id, String nom, String description, String categorie, List<Professional> professionnels, String service_photo) {
        this.id = id;
        this.nom = nom;
        this.description = description;
        this.categorie = categorie;
        this.professionnels = professionnels;
        this.service_photo = service_photo;
    }

    public Service() {
    }

    public Long getId() {
        return id;
    }

    public String getNom() {
        return nom;
    }

    public String getDescription() {
        return description;
    }

    public String getCategorie() {
        return categorie;
    }

    public List<Professional> getProfessionnels() {
        return professionnels;
    }

    public String getService_photo() {
        return service_photo;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setCategorie(String categorie) {
        this.categorie = categorie;
    }

    public void setProfessionnels(List<Professional> professionnels) {
        this.professionnels = professionnels;
    }

    public void setService_photo(String service_photo) {
        this.service_photo = service_photo;
    }
}
